controllers/openstackdataplanenode_controller.go

```diff
@@ -114,30 +103,11 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 		delete(instance.ObjectMeta.Labels, "openstackdataplanerole")
 	}
 
-	instanceRole, err := r.GetInstanceRole(ctx, instance)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-
-	ansibleSSHPrivateKeySecret := r.GetAnsibleSSHPrivateKeySecret(instance, instanceRole)
-	_, result, err = secret.VerifySecret(
-		ctx,
-		types.NamespacedName{Namespace: instance.Namespace, Name: ansibleSSHPrivateKeySecret},
-		[]string{
-			"ssh-privatekey",
-		},
-		r.Client,
-		time.Duration(5)*time.Second,
-	)
-	if err != nil {
-		return result, err
-	}
-
 	// Always patch the instance status when exiting this function so we can
 	// persist any changes.
 	defer func() {
 		// update the overall status condition if service is ready
-		if instance.IsReady() || instanceRole.IsReady() {
+		if instance.IsReady() {
```

Comment:
As we're not reconciling the node separately, we probably don't need to. We can keep each node provisioning/deployment status in role rather than in the respective node CRDs. We can also update the node status from the role controller, but that won't be a good design. 

I'm working on  a patch to simplify the status/conditions  which I'll add to the PR.